sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -598,26 +531,26 @@ object IntervalUtils {
                 } else if (s.matchAt(microsStr, i)) {
                   microseconds = Math.addExact(microseconds, currentValue)
                   i += microsStr.numBytes()
-                } else return null
-              case _ => return null
+                } else throwIAE(s"invalid unit '$nextWord'")
+              case _ => throwIAE(s"invalid unit '$nextWord'")
             }
           } catch {
-            case _: ArithmeticException => return null
+            case e: ArithmeticException => throwIAE(e.getMessage, e)
           }
           state = UNIT_SUFFIX
         case UNIT_SUFFIX =>
           b match {
             case 's' => state = UNIT_END
             case ' ' => state = TRIM_BEFORE_SIGN
-            case _ => return null
+            case _ => throwIAE(s"invalid unit suffix '$nextWord'")
```

Comment:
`invalid unit '$nextWord'` is better if we can implement `nextword` correctly. Or we should introduce "currentWord"

Reply:
the `nextword ` represents which character the error occurs, the only special case is for nanoseconds out of range, the fraction + `nextword` could just exactly show the out of range number

Reply:
or we change the logic of unit parsing we can extract the whole part like `case  _ if b>= 'A' $$ b<= 'z' => unit = s.substring(i, s.numBytes()).subStringIndex(UTF8String.blankString(1), 1)` than do `unit` case matching and error capture.

Reply:
for error reporting, usually backtracing is necessary. For example, it's better to tell users that `123a` is not valid, instead of just saying `a` is not valid.